pkg/metrics_pusher/pusher.go

```diff
@@ -62,23 +64,30 @@ func (p *MetricsPusher) pushMetrics() {
 			return
 		}
 	} else {
+		var resp *http.Response
 		caCert, err := ioutil.ReadFile(p.Clientca)
-		if err != nil {
-			p.Log.Error(err, "Unable to read CA cert for fetching metrics: Please provide valid CA cert or path")
+		if errors.Is(err, os.ErrNotExist) {
+			// Path for no CA cert given (or file location doesn't exist)
+			p.Log.Info("Unable to read CA cert file for fetching metrics, going to assuming none are needed.")
```

Comment:
We should not make this assumption.  If a valid file is specified then it must be found.  Otherwise the caller will have assumed that metrics pusher was setup with a valid client CA.  I think the proper solution is to simply skip the file read if `--client-ca-file` is empty.  Otherwise, we must assume that the caller desires `--client-ca-file` to be used.